app.test.js

```diff
@@ -108,12 +107,114 @@ describe("GET: /api/articles", () => {
+    expect((body.article.length)).toBe(2)
+    expect((body.article[0].article_id)).toBe(9)
+    expect((body.article[1].article_id)).toBe(9)
+    expect(body.article).toBeSortedBy('created_at', {descending: true})
+  })
+})
+  test("Return 404 error if article ID doesnt exist", () => {
+    return request(app)
+    .get('/api/articles/420420/comments')
+    .expect(404)
+    .then((body) => {
+      expect(body.notFound).toBe(true)
+    })
+  })
+  
+  describe("POST: /api/articles/:article_id/comments", () => {
+    test("Return new comment to specified article", () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "icellusedkars"})
+      .expect(201)
+      .then(({body}) => {
+        expect((expect((body.body)).toBe("Hello")))
+        expect((body.author)).toBe("icellusedkars")
+      })
+    })
+    test("Return not found if id is incorrect", () => {
+      return request(app)
+      .post('/api/articles/3945/comments')
+      .send({body: "Hello", username: "icellusedkars"})
+      .expect(400)
+      .then(({body}) => {
+      expect(body.msg).toBe('not an id')
+    })
+    })
+    test("Ensure unnessiary properties are ignored", () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "icellusedkars", chicken: "waffle"})
+      .expect(201)
+      .then(({body}) => {
+        expect(body).toMatchObject( {
+          comment_id: 19,
+          body: 'Hello',
+          article_id: 9,
+          author: 'icellusedkars',
+          votes: 0
+        })
+        expect(body.chicken).toBe(undefined)
+      })
+    })
+    test('Return 404 if username invalid', () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "Marlon-Brando"})
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe('Username doesnt exist')
+      })
+    })
+  })
+})
+describe("PATCH: /api/articles/:article_id", () => {
+  test('Return updated article votes', () => {
+    return request(app)
+          .patch('/api/articles/1')
+          .send({inc_votes : -33})
+          .expect(200)
+          .then((data) => {
+            expect(data.body).toMatchObject({
+              article_id: 1,
+              title: 'Living in the shadow of a great man',
+              topic: 'mitch',
+              author: 'butter_bridge',
+              body: 'I find this existence challenging',
+              votes: 67,
+              article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+            })
+          })
+        })
+  test('Return 404 if article ID invalid', () => {
+      return request(app)
+        .patch('/api/articles/93939')
+        .send({inc_votes : -33})
+        .expect(404)
+        .then(({body}) => {
+          expect(body.msg).toBe('Article not found')
+            })
+          })
+
+
```

Comment:
Great tests written here, however there a few missing 👀 Please add the following:

Status 200 - would increment the current review's vote property by 1
Status 400, incorrect body, e.g. inc_votes property is not a number
Status 400, invalid ID, e.g. string of "not-an-id"